[Merged by Bors] - Addition to Lighthouse Book faq.md

chong-he opened the pull request:
## Issue Addressed

Added some frequently asked questions in the Lighthouse Book

## Proposed Changes

Created another file faqV2.md which categorises the FAQs into different sections for better organisation. For review purpose, a review on faq.md will suffice. Then, if faqV2.md looks better, can delete faq.md; otherwise if the changes to the faqV2.md is too much, can keep faq.md and delete faqv2.md.

## Additional Info

Please provide any additional information. For example, future considerations
or information useful for reviewers.

chong-he committed:
Update FAQ

chong-he committed:
Revised and update FAQ

chong-he committed:
Fix a minor error

chong-he committed:
Update faq

chong-he committed:
Minor correction

chong-he committed:
Update faqv2

chong-he committed:
Update FAQ

chong-he committed:
Update FAQ

paulhauner commented on book/src/faqV2.md:
```diff
@@ -0,0 +1,487 @@
+# Frequently Asked Questions
+
+## [Beacon Node](#beacon-node-1)
+- [I see warning about "Syncing deposit contract block cache" or error about "updating deposit contract cache", what should I do?](#i-see-warning-about-syncing-deposit-contract-block-cache-or-error-about-updating-deposit-contract-cache-what-should-i-do)
```
```suggestion
- [I see a warning about "Syncing deposit contract block cache" or an error about "updating deposit contract cache", what should I do?](#i-see-a-warning-about-syncing-deposit-contract-block-cache-or-an-error-about-updating-deposit-contract-cache-what-should-i-do)
```

paulhauner commented on book/src/faqV2.md:
```diff
@@ -0,0 +1,487 @@
+- [I see beacon logs showing `WARN: Execution engine called failed`, what should I do?](#i-see-beacon-logs-showing-warn-execution-engine-called-failed-what-should-i-do)
+- [My beacon node is stuck at downloading historical block using checkpoint sync. What should I do?](#my-beacon-node-is-stuck-at-downloading-historical-block-using-checkpoint-sync-what-should-i-do)
+- [I proposed a block but the beacon node shows `could not publish message` with error `duplicate` as below, should I be worried?](#i-proposed-a-block-but-the-beacon-node-shows-could-not-publish-message-with-error-duplicate-as-below-should-i-be-worried)
+- [I see beacon node logs `Head is optimistic`, and I am missing attestations. What should I do?](#i-see-beacon-node-logs-head-is-optimistic-and-i-am-missing-attestations-what-should-i-do)
```
```suggestion
- [I see beacon node logs `Head is optimistic` and I am missing attestations. What should I do?](#i-see-beacon-node-logs-head-is-optimistic-and-i-am-missing-attestations-what-should-i-do)
```

paulhauner commented on book/src/faqV2.md:
```diff
@@ -0,0 +1,487 @@
+- [Can I submit a voluntary exit message without a beacon node?](#can-i-submit-a-voluntary-exit-message-without-running-a-beacon-node)
+- [Does increasing the number of validators increase the CPU and other computer resources used?](#does-increasing-the-number-of-validators-increase-the-cpu-and-other-computer-resources-used)
+- [I want to add new validators. Do I have to reimport the existing keys?](#i-want-to-add-new-validators-do-i-have-to-reimport-the-existing-keys)
+- [Do I have to stop `lighthouse vc` when import new validator keys?](#do-i-have-to-stop-lighthouse-vc-when-import-new-validator-keys)
```
```suggestion
- [Do I have to stop `lighthouse vc` the when importing new validator keys?](#do-i-have-to-stop-lighthouse-vc-when-importing-new-validator-keys)
```

paulhauner commented on book/src/faqV2.md:
```diff
@@ -0,0 +1,487 @@
+
+## Beacon Node
+
+### I see warning about "Syncing deposit contract block cache" or error about "updating deposit contract cache", what should I do?
```
```suggestion
### I see a warning about "Syncing deposit contract block cache" or an error about "updating deposit contract cache", what should I do?
```

paulhauner commented on book/src/faqV2.md:
````diff
@@ -0,0 +1,487 @@
+WARN BlockProcessingFailure                  outcome: ValidatorPubkeyCacheLockTimeout, msg: unexpected condition in processing block.
+```
+
+A `Timeout` error suggests that the computer may be overloading at the moment, for example, the execution client is still syncing. You may use the flag `--disable-lock-timeouts` to silence this error, although it will not fix the underlying slowness. Nevertheless, this is a relatively harmless log, and the error should go away once the resources used are back to normal. 
````
```suggestion
A `Timeout` error suggests that the computer may be overloaded at the moment, for example, the execution client is still syncing. You may use the flag `--disable-lock-timeouts` to silence this error, although it will not fix the underlying slowness. Nevertheless, this is a relatively harmless log, and the error should go away once the resources used are back to normal. 
```

paulhauner commented on book/src/faqV2.md:
````diff
@@ -0,0 +1,487 @@
+WARN BlockProcessingFailure                  outcome: MissingBeaconBlock(0xbdba211f8d72029554e405d8e4906690dca807d1d7b1bc8c9b88d7970f1648bc), msg: unexpected condition in processing block.
+```
+
+`MissingBeaconBlock` suggests that the database has corrupted. You should wide the database and use [Checkpoint Sync](./checkpoint-sync.md) to resync the beacon chain. 
````
```suggestion
`MissingBeaconBlock` suggests that the database has corrupted. You should wipe the database and use [Checkpoint Sync](./checkpoint-sync.md) to resync the beacon chain. 
```

paulhauner commented on book/src/faqV2.md:
```diff
@@ -0,0 +1,487 @@
+
+### After checkpoint sync, the progress of `downloading historical blocks` is slow. Why?
+
+This is a normal behaviour. Since [v4.1.0](https://github.com/sigp/lighthouse/releases/tag/v4.1.0), Lighthouse implements rate-limited backfill sync to mitigate validator performance issues after a checkpoint sync. This is not something to worry because backfill sync / historical data is not required for staking. However, if you opt to sync the chain as fast as possible, you can add the flag `--d ...
```
```suggestion
This is a normal behaviour. Since [v4.1.0](https://github.com/sigp/lighthouse/releases/tag/v4.1.0), Lighthouse implements rate-limited backfill sync to mitigate validator performance issues after a checkpoint sync. This is not something to worry about since backfill sync / historical data is not required for staking. However, if you opt to sync the chain as fast as possible, you can add the flag `--disable-backfill-rate-limiting` to the beacon node.
```

paulhauner commented on book/src/faqV2.md:
```diff
@@ -0,0 +1,487 @@
+
+This warning usually comes with an http error code. Some examples are given below:
+
+1. The log show:
```
```suggestion
1. The log shows:
```

paulhauner commented on book/src/faqV2.md:
```diff
@@ -0,0 +1,487 @@
+
+The error is `500 Internal Server Error`. This suggests that the execution client is not synced. Once the execution client is synced, the error will disappear.
+
+2. The log show:
```
```suggestion
2. The log shows:
```

paulhauner commented on book/src/faqV2.md:
````diff
@@ -0,0 +1,487 @@
+ERRO Failed to download attester duties      err: FailedToDownloadAttesters("Some endpoints failed, num_failed: 2 http://localhost:5052/ => Unavailable(NotSynced), http://localhost:5052/ => RequestFailed(ServerMessage(ErrorMessage { code: 503, message: \"SERVICE_UNAVAILABLE: beacon node is syncing
+```
+
+This means that the validator client is sending requests to the beacon node. However, because the beacon node is still syncing, hence it is unable to fulfil the request. The error will disappear once the beacon node is synced. 
````
```suggestion
This means that the validator client is sending requests to the beacon node. However the beacon node is still syncing is therefore unable to fulfil the request. The error will disappear once the beacon node is synced. 
```

paulhauner commented on book/src/faqV2.md:
```diff
@@ -0,0 +1,487 @@
+
+### Sometimes I miss the attestation head vote, resulting in penalty. Is this normal?
+
+In general, it is unavoidable to have some penalties occasionally. This is particularly the case when you are assigned to attest on the first slot of an epoch and if the proposer of that slot releases the block late, then you will get penalised for missing the target and head votes. Your attestation performance does not only depend on your own setup, but also on everyone else's performance.
```
```suggestion
In general, it is unavoidable to have some penalties occasionally. This is particularly the case when you are assigned to attest on the first slot of an epoch and if the proposer of that slot releases the block late, then you will get penalised for missing the target and head votes. Your attestation performance does not only depend on your own setup, but also on everyone elses performance.
```

paulhauner commented on book/src/faqV2.md:
```diff
@@ -0,0 +1,487 @@
+
+No. You can just import new validator keys to the destination directory. If the `validator_keys` folder contains existing keys, that's fine as well because Lighthouse will skip importing existing keys.
+
+### Do I have to stop `lighthouse vc` when import new validator keys?
```
```suggestion
### Do I have to stop `lighthouse vc` when importing new validator keys?
```

paulhauner reviewed (changes requested):
Nice work! I really like the `faqV2.md` layout, so I'd suggest replacing the existing `faq.md` file with that.

I gave `faqV2.md` a full read through and fixed some typos. I think some of them were there before your changes, but now seems like a good time to fix them :)

I can see the links to chapters are pretty big and cumbersome, I'll share another way to go about it using [named anchors](https://stackoverflow.com/a/7015050) in case you're interested:

## Code:

```markdown
- [A big sentence with a big link](#a-big-sentence-with-a-big-link)
- [A big sentence with a little link](#little-link)

## A big sentence with a big link

Whenever we change this heading we need to make sure we update the link too.

<a name="little-link"></a>
## A big sentence with a little link

We can change the name of this heading without changing the link name.
```

paulhauner commented:
Could you please remove the `waiting-on-author` tag and add the `ready-for-review` when it's ready for review again please? ☺️

chong-he committed:
Apply suggestions from code review

Co-authored-by: Paul Hauner <paul@paulhauner.com>

chong-he committed:
Name anchor

chong-he committed:
Removed faq and renamed faqV2 to faq

chong-he commented:
> Nice work! I really like the `faqV2.md` layout, so I'd suggest replacing the existing `faq.md` file with that.
> 
> I gave `faqV2.md` a full read through and fixed some typos. I think some of them were there before your changes, but now seems like a good time to fix them :)
> 
> I can see the links to chapters are pretty big and cumbersome, I'll share another way to go about it using [named anchors](https://stackoverflow.com/a/7015050) in case you're interested:
> 
> ## Code:
> ```gfm
> - [A big sentence with a big link](#a-big-sentence-with-a-big-link)
> - [A big sentence with a little link](#little-link)
> 
> ## A big sentence with a big link
> 
> Whenever we change this heading we need to make sure we update the link too.
> 
> <a name="little-link"></a>
> ## A big sentence with a little link
> 
> We can change the name of this heading without changing the link name.
> ```

Thanks for review! I have committed to the suggestions, ~deleted faq.mv and renamed faqV2 to faq.~ 
Edit: I figured out it's probably better to keep faqV2 as it is so that the changes can be seen, so I modifed the Summary.md instead

Also thanks for the heads-up on named anchor. I have applied it, they look better now

chong-he committed:
Revert "Removed faq and renamed faqV2 to faq"

This reverts commit 5aa41f3c8a486978ef2f83682626ddb3ae490a67.

chong-he committed:
Update summary and delete faq

chong-he committed:
delete faq

paulhauner commented:
Very nice! Would you be open to renaming `faqV2.md` to just `faq.md` please? I think it's a little cleaner that way :)

I'm happy to merge after that's done!

chong-he committed:
Rename faqv2 to faq

chong-he commented:
> Very nice! Would you be open to renaming `faqV2.md` to just `faq.md` please? I think it's a little cleaner that way :)
> 
> I'm happy to merge after that's done!

Done :)

paulhauner reviewed (approved):
Perfect!

paulhauner commented:
bors r+

paulhauner commented:
bors r+

paulhauner commented:
bors r-

paulhauner commented:
bors r+

paulhauner commented:
bors r+

michaelsproul commented:
bors r-
bors r+
